test/unit/config_test.rb

```diff
@@ -0,0 +1,417 @@
+      let(:fixture_name) { :like_legacy }
+      it 'should read the options' do
+        expected = ['color', 'reporter', 'target_id', 'type'].sort
+        seen_fields.must_equal expected
+        final_options['color'].must_equal "true"  # Dubious
+        final_options['target_id'].must_equal 'mynode'
+      end
+    end
+
+    describe 'when the CLI opts are present in a legacy file' do
+      let(:fixture_name) { :legacy }
+      it 'should read the options' do
+        expected = ['color', 'reporter', 'target_id', 'type'].sort
+        seen_fields.must_equal expected
+        final_options['color'].must_equal "true"  # Dubious
+        final_options['target_id'].must_equal 'mynode'
+      end
+    end
+  end
+
+  # ========================================================================== #
+  #                     Parsing and Validating Reporters
+  # ========================================================================== #
+
+  # TODO: this should be moved into plugins for the reporters
+  describe 'when parsing reporters' do
+    let(:cfg) { Inspec::Config.new(cli_opts) }
+    let(:seen_reporters) { cfg['reporter'] }
+
+    describe 'when paring CLI reporter' do
+      let(:cli_opts) { { 'reporter' => ['cli'] } }
+      it 'parse cli reporters' do
+        expected_value = { 'cli' => { 'stdout' => true }}
+        seen_reporters.must_equal expected_value
+      end
+    end
+
+    describe 'when paring CLI reporter' do
+      let(:cli_opts) { { 'reporter' => ['cli'], 'target_id' => '1d3e399f-4d71-4863-ac54-84d437fbc444' } }
+      it 'parses cli report and attaches target_id' do
+        expected_value = {"cli"=>{"stdout"=>true, "target_id"=>"1d3e399f-4d71-4863-ac54-84d437fbc444"}}
+        seen_reporters.must_equal expected_value
+      end
+    end
+  end
+
+  describe 'when validating reporters' do
+    # validate_reporters is private, so we use .send
+    let(:cfg) { Inspec::Config.new }
+    it 'valid reporter' do
+      reporters = { 'json' => { 'stdout' => true } }
+      cfg.send(:validate_reporters, reporters)
+    end
+
+    it 'invalid reporter type' do
+      reporters = ['json', 'magenta']
+      proc { cfg.send(:validate_reporters, reporters) }.must_raise NotImplementedError
+    end
+
+    it 'two reporters outputting to stdout' do
+      stdout = { 'stdout' => true }
+      reporters = { 'json' => stdout, 'cli' => stdout }
+      proc { cfg.send(:validate_reporters, reporters) }.must_raise ArgumentError
+    end
+  end
+
+  # ========================================================================== #
+  #                      Miscellaneous Option Finalization
+  # ========================================================================== #
+
+  describe 'option finalization' do
+    it 'raises if `--password/--sudo-password` are used without value' do
+      # When you invoke `inspec shell --password`  (with no value for password,
+      # though it is setup to expect a string) Thor will set the key with value -1
```

Comment:
See other comment about context and 

https://github.com/inspec/inspec/pull/1918/files#diff-2b20044866abd29a7a781f5760238403R30

Reply:
This is just copied from the base_cli_test.rb

https://github.com/inspec/inspec/blob/feefdcf9afba80f4e666e6dad5ffce6bd30019dd/test/unit/base_cli_test.rb#L11